synapse/http/servlet.py

```diff
@@ -108,6 +110,60 @@ def parse_boolean_from_args(args, name, default=None, required=False):
             return default
 
 
+@overload
+def parse_bytes(
+    request: Request,
+    name: str,
+    default: Literal[None] = None,
+    required: Literal[True] = True,
+) -> bytes:
+    ...
+
+
+@overload
+def parse_bytes(
+    request: Request,
+    name: str,
+    default: Optional[bytes] = None,
+    required: bool = False,
+) -> Optional[bytes]:
+    ...
+
+
+def parse_bytes(
```

Comment:
`parse_bytes` seems like a rubbish name to me (as does `parse_string`).

Suggestion would be that you replace it with `parse_bytes_from_args` and make it take an `args` dict (and do `request.args` in the callers).

Reply:
That seems fine with me! I was mostly matching `parse_string` (which should maybe be `parse_string_from_request`, but it really is just `parse_string_from_args`...)

Reply:
I pushed a commit, it does need some typing foo since `request.args` pops up as `Optional[Any]`, I think this is probably OK though?

Another option would be to do `parse_bytes_from_request` (or `parse_bytes_from_request_args`) and have it still take the `Request` object.

Reply:
yet another option might be to allow `parse_bytes_from_request` to take `None` and do something plausible?